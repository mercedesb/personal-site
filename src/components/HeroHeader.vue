<template>
  <main>
    <div :class="'Hero Hero--' + color">
      <h2>{{ title }}</h2>
    </div>
  </main>
</template>

<script>
export default {
  props: {
    title: String,
    color: String
  }
}
</script>

<style lang="scss">
</style>