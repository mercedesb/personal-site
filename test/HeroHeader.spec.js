import { shallow } from '@vue/test-utils'
import HeroHeader from '@/components/HeroHeader.vue'

describe('HeroHeader', () => {
  describe('with a title', () => {
    it('matches snapshot', () => {
      const wrapper = shallow(HeroHeader, {
        propsData: {
          color: 'brown',
          title: 'title',
          mainImage: 'http://image.com',
          backgroundImages: ['http://image1.com', 'http://image2.com', 'http://image3.com']
        }
      })
      expect(wrapper.html()).toMatchSnapshot()
    })
  })

  describe('without a title', () => {
    it('matches snapshot', () => {
      const wrapper = shallow(HeroHeader, {
        propsData: {
          color: 'brown',
          media: 'http://image.com',
          backgroundImages: ['http://image1.com', 'http://image2.com', 'http://image3.com']
        }
      })
      expect(wrapper.html()).toMatchSnapshot()
    })
  })
})