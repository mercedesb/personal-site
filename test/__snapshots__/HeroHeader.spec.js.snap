// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeroHeader with a title matches snapshot 1`] = `
<header class="Hero Hero--brown">
  <div class="Hero-background">
    <img src="http://image1.com" class="Hero-backgroundImage">
    <img src="http://image2.com" class="Hero-backgroundImage">
    <img src="http://image3.com" class="Hero-backgroundImage">
  </div>
  <div class="Hero-contentContainer">
    <div>
      <img src="http://image.com" class="Hero-image">
      <parse-markdown source="title"></parse-markdown>
    </div>
  </div>
</header>
`;

exports[`HeroHeader without a title matches snapshot 1`] = `undefined`;
